Update CC goal documentation (#9678)
strimzi/strimzi-kafka-operator: documentation/modules/cruise-control/con-optimization-goals.adoc

== Hard and soft optimization goals

Hard goals are goals that _must_ be satisfied in optimization proposals.

> Goals that are not hard coded as _hard goals_ in Cruise Control are known as _soft goals_.

Reviewer: ```suggestion
Goals that are not defined as _hard goals_ in Cruise Control code are known as _soft goals_.
```

NOTE: For example, you might have a soft goal to distribute a topic's replicas evenly across the cluster (the topic replica distribution goal).
Cruise Control will ignore this goal if doing so enables all the configured hard goals to be met.

> In Cruise Control, the following xref:main-goals[main optimization goals] are hard coded as hard goals:

Reviewer: ```suggestion
In Cruise Control, the following xref:main-goals[main optimization goals] are hard goals:
```

[source]
RackAwareGoal; ReplicaCapacityGoal; DiskCapacityGoal; NetworkInboundCapacityGoal; NetworkOutboundCapacityGoal; CpuCapacityGoal

> You configure which hard goals will be enforced to execute in the Cruise Control deployment configuration, by editing the `hard.goals` property in `Kafka.spec.cruiseControl.config`.

Reviewer 2: ```suggestion
In your Cruise Control deployment configuration, you can specify which hard goals to enforce using the `hard.goals` property in `Kafka.spec.cruiseControl.config`.
```

> * To enforce execution of all hard goals, simply omit the `hard.goals` property."

Reviewer 2: ```suggestion
* To enforce execution of all hard goals, simply omit the `hard.goals` property.
```

Reviewer 2: Sorry about that!

> * To enforce that all the hard goals are executed, do not specify the `hard.goals` property in `Kafka.spec.cruiseControl.config`

Reviewer 2: ```suggestion
* To enforce execution of all hard goals, simply omit the `hard.goals` property."
```

> * To change which hard goals are enforced to execute by Cruise Control, specify the desired goals in the `hard.goals` property, using their fully-qualified domain names.

Reviewer 2: ```suggestion
* To change which hard goals Cruise Control enforces, specify the required goals in the `hard.goals` property using their fully-qualified domain names.
```

> 
> * To prevent execution of a specific hard goal, ensure that the goal is not included in both the `default.goals` and `hard.goals` list configurations.

Reviewer 2: ```suggestion
* To prevent execution of a specific hard goal, ensure that the goal is not included in both the `default.goals` and `hard.goals` list configurations.
```

> * To prevent a hard goal from being executed, make sure the goal is not included in the `default.goals` and `hard.goals` list configurations.

Reviewer 2: ```suggestion
* To prevent execution of a specific hard goal, ensure that the goal is not included in both the `default.goals` and `hard.goals` list configurations.
```

> NOTE: We can not configure which goals are _soft goals_ or _hard goals_.
> This is defined by the Cruise Control code.

Reviewer 2: ```suggestion
NOTE: It is not possible to configure which goals are considered soft or hard goals. 
This distinction is determined by the Cruise Control code.
```
